### DISCUSSION TITLE
Button layer auto activation

### DISCUSSION THREAD

**Discussion by volderette** [manna-harbour/miryoku#166] [Q&A] [answered]
Hi!

I read, that the button layer is available for auto activation? How would I do that? I'm using a charybdis keyboard and I tried to copy over code from their default keymap, but it seems when I try to set any layer it's not available.

Thanks for your help and awesome work!

**Comment by manna-harbour** [comment, marked as answer]
Unfortunately QMK doesn't support adding `#include "users/manna-harbour_miryoku/manna-harbour_miryoku.h"` to `config.h` to include the layer enums because it also uses `config.h` to assemble assembler and the enums aren't valid there. Instead, you have to manually add a define for the layer number to `config.h` then use that macro when specifying the layer for auto activation. By default Button is layer 3, so e.g. add `#define BUTTON_LAYER 3` and then use `BUTTON_LAYER` in your configuration. Of course you could also use the bare layer number directly and add a comment.

**Reply by volderette** [reply]
Thank you!

**Comment by PiousAeneas** [comment]
@manna-harbour Sorry if this is a silly question, but would moving the mouse when in the Miryoku Mouse Layer trigger the Button layer if using QMK's [Automatic Mouse Layer](https://docs.qmk.fm/features/pointing_device#pointing-device-auto-mouse) functionality or does the automatic mouse layer somehow only trigger when it detects e.g. an optical sensor mouse? Just trying to understand how this might work since I like the idea of being able to use a Ploopy Nano with my Miryoku layout.

**Reply by manna-harbour** [reply]
No, the keyboard doesn't know anything about the mouse on the host. You'd need a communication channel from mouse to keyboard. There are a few different ways to achieve this, but they require custom keyboard firmware, and either custom mouse firmware or a custom utility on the host.

**Reply by robrecord** [reply]
I am in the same situation with the same question. Do you have any leads as to what to search for? The Ploopy Nano has QMK firmware, but unsure what feature needs turning on or adding in, or what the host utility would be. Do you know what search terms we might need to use to find these?

**Reply by manna-harbour** [reply]
For a QMK mouse, the easiest way is to set the state of an indicator LED such as Scroll Lock on the mouse based on movement, and activate the Button layer based on that. Similar to this, but in the opposite direction: https://www.reddit.com/r/ploopy/comments/nlvgkq/how_to_scroll_with_the_trackball_nano/. E.g. https://www.reddit.com/r/ErgoMechKeyboards/comments/xtd6qb/comment/iqqnt63.
